metal/uart.h

```diff
@@ -33,11 +34,35 @@ struct metal_uart_vtable {
     size_t (*get_rx_watermark)(struct metal_uart *uart);
 };
 
+/*!
+ * @brief Enum type for describing the cause of a UART callaback
+ */
+typedef enum {
+    METAL_UART_TX_DONE,
+    METAL_UART_RX_DONE,
```

Comment:
What do you think about making the `MIN`, `MAX`, `UNKNOWN`, and `COUNT` values "private" by prefixing them with `_`? They seem like internal values for API testing, not values we expect the end user to consume.

Reply:
To be honest, I used to use such "values" ... depending on enumeration kind ...
For first implementation, it could be private ... but I encourage you to keep this in mind, especially if/when FAEs are going to come back to us asking for such changes in drivers ...
I understand it's huge work to update every drivers.